pkg/controller/cover/cover.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cover implements the cover traffic API.
+package cover
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"math/big"
+	"net/http"
+	"time"
+
+	"github.com/google/exposure-notifications-verification-server/pkg/api"
+	"github.com/google/exposure-notifications-verification-server/pkg/config"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+	"github.com/google/exposure-notifications-verification-server/pkg/logging"
+
+	"go.uber.org/zap"
+)
+
+const (
+	minDelayMS  = 250
+	varWindowMS = 500
+
+	minRandomBytes = 400
+	varBytesWindow = 200
+)
+
+// CoverAPI is an http.Handler for servering CoverRequest API calls.
+type CoverAPI struct {
+	config *config.APIServerConfig
+	logger *zap.SugaredLogger
+}
+
+// New creates a new CoverAPI.
+func New(ctx context.Context, config *config.APIServerConfig) http.Handler {
+	return &CoverAPI{config, logging.FromContext(ctx)}
+}
+
+func (c *CoverAPI) randomDelay() {
+	delay, err := rand.Int(rand.Reader, big.NewInt(int64(varWindowMS)))
+	if err != nil {
+		c.logger.Errorf("error getting random int, using min sleep: %v", err)
+		delay = big.NewInt(0)
+	}
+
+	length := time.Duration(minDelayMS+delay.Int64()) * time.Millisecond
+	time.Sleep(length)
+}
+
+func (c *CoverAPI) randomData() (string, error) {
```

Comment:
Is there any way we can ensure these get configured well? If they aren't accurate to the live payload, the chaff is easily detected as chaff.

Reply:
Yes - will send a followup that tracks and normalizes both request size and latency

https://github.com/mikehelmick/go-chaff

Reply:
added necessary TODOs